### DISCUSSION TITLE
How add to select(), field with const string?

### DISCUSSION THREAD

**Discussion by qarmin** [diesel-rs/diesel#3861] [Q&A]
```
    let mut inner_reports_sql = report_file_table
        .filter(report_file::origin.eq(ReportOrigin::HFA700.to_string()))
        .filter(report_file::patient_uuid.eq(patient_uuid))
        .select((
            report_file::id,
            "Imported Record".into()
        ))
        .load::<CustomPatientReportFileSql>(conn)?;
```
How to add as second item, `Imported Record` string?

EDIT: 
Nevermind - before I tried to find answer ~1h, but after posting this question I found solution in 5 minutes (maybe this should be added to docs here? - https://docs.diesel.rs/2.1.x/diesel/prelude/trait.QueryDsl.html#method.select)
```
    let mut inner_reports_sql = report_file_table
        .filter(report_file::origin.eq(ReportOrigin::HFA700.to_string()))
        .filter(report_file::patient_uuid.eq(patient_uuid))
        .select((
            report_file::id,
            sql::<Text>("'Imported Record'"),
        ))
        .load::<CustomPatientReportFileSql>(conn)?;
```

**Comment by weiznich** [comment]
Using `dsl::sql` is something like a shortcut that always works, however it has a few downsides. Notably:

* It prevents caching the prepared statement as we don't know whether or not the internal string is unsafe to cache
* It opens up the possibility for sql injections if you pass a user supplied string as argument to that function

In this particular case you can use [`IntoSql::into_sql`](https://docs.diesel.rs/2.1.x/diesel/expression/trait.IntoSql.html#method.into_sql) to workaround that limitations. That will use a bind value to provide the constant value, which then can be used in a select clause (or everywhere else in the query dsl).
